ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -229,28 +228,30 @@ private void cleanupExpiredInflightRequests() {
   private void processServerError(ReplicaId replica, ServerErrorCode serverErrorCode) {
     switch (serverErrorCode) {
       case No_Error:
-        logger.trace("The delete request was successful.");
         operationTracker.onResponse(replica, true);
         break;
       case Blob_Deleted:
-        logger.trace("Blob has already been deleted.");
         operationTracker.onResponse(replica, true);
         break;
       case Blob_Expired:
         updateOperationState(replica, RouterErrorCode.BlobExpired);
         break;
       case Blob_Not_Found:
-      case Partition_Unknown:
         updateOperationState(replica, RouterErrorCode.BlobDoesNotExist);
         break;
+      case Partition_Unknown:
+        updateOperationState(replica, RouterErrorCode.UnexpectedInternalError);
```

Comment:
The coordinator is logging this as BlobDoesNotExist. I am inclined to your change, but just make sure what was the reason for the legacy code, and this change would not cause any inconsistency.